### ISSUE TITLE
Exception occurs when logging out from authorized connection

### ISSUE THREAD

**Issue by aalmiray** [canoo/dolphin-platform#73]
We have a spring-boot server with security enabled. The Java client connects to the server once the connection has been authorized (we perform some cookie handling in order for this to work). Everything works fine during normal operations. However we get the following exception when the logout procedure is invoked (via clientContext.disconnect()):

```
org.apache.http.client.HttpResponseException: Unauthorized
    at org.opendolphin.core.client.comm.SessionAffinityCheckingResponseHandler.handleResponse(HttpClientConnector.groovy:109)
    at org.opendolphin.core.client.comm.SessionAffinityCheckingResponseHandler.handleResponse(HttpClientConnector.groovy)
    at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:222)
    at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:164)
    at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:139)
    at org.apache.http.client.HttpClient$execute.call(Unknown Source)
    at org.opendolphin.core.client.comm.HttpClientConnector.transmit(HttpClientConnector.groovy:81)
    at org.opendolphin.core.client.comm.ClientConnector$_startCommandProcessing_closure2_closure10.doCall(ClientConnector.groovy:76)
    at org.opendolphin.core.client.comm.ClientConnector$_startCommandProcessing_closure2_closure10.doCall(ClientConnector.groovy)
    at org.opendolphin.core.client.comm.ClientConnector.doExceptionSafe(ClientConnector.groovy:145)
    at org.opendolphin.core.client.comm.ClientConnector$_startCommandProcessing_closure2.doCall(ClientConnector.groovy:78)
    at groovyx.gpars.dataflow.operator.DataflowOperatorActor.startTask(DataflowOperatorActor.java:120)
    at groovyx.gpars.dataflow.operator.DataflowOperatorActor.onMessage(DataflowOperatorActor.java:108)
    at groovyx.gpars.actor.impl.SDAClosure$1.call(SDAClosure.java:43)
    at groovyx.gpars.actor.AbstractLoopingActor.runEnhancedWithoutRepliesOnMessages(AbstractLoopingActor.java:293)
    at groovyx.gpars.actor.AbstractLoopingActor.access$400(AbstractLoopingActor.java:30)
    at groovyx.gpars.actor.AbstractLoopingActor$1.handleMessage(AbstractLoopingActor.java:93)
    at groovyx.gpars.util.AsyncMessagingCore.run(AsyncMessagingCore.java:132)
```

The problem appears to be that the Java client continues to issue requests to the server (long polling perhaps?) and fails because it's connection is no longer authorized.

**Comment by hendrikebbers** [comment]
Is this exception thrown 1 or several times?

Von meinem iPhone gesendet

**Comment by netopyr** [comment]
How comes the connection is no longer authorized? In the web client we have to explicit make a request to .../logout, which is handled by Spring and kills the session. Are we doing this automatically in the Java client?

**Comment by hendrikebbers** [comment]
Not in the DP Java client. Maybe it’s part of the application?

**Comment by aalmiray** [comment]
No, we don't call /logout from the Java client. We simply call `clientContext.disconnect()`. That should stop all communication between client and server. 

In order to call /logout form the Java client one must create an instance of HttpClient; copy the authcookie from the DP/OD HttpClient into this new client; then issue a request for /logout.

**Comment by hendrikebbers** [comment]
Maybe this is the long running poll that is executed when disconnect() is called. The request was already send in that case, then you call disconnect() that stops polling (but one poll request is still "on the server"). Then the real security disconnect (/logout) is done. By doing so the response of the poll request will contain the "Unauthorized". But this is just an idea without looking in the code. If this is right then we need to disable polling first and than wait for the polling response before we can say that everything is disconnected. Mmmh, the big question is how we can organize this.

**Comment by hendrikebbers** [comment]
maybe this is based on #92 ?

**Comment by hendrikebbers** [comment]
@aalmiray Argh, I can't find time to get into this issue. Maybe you can have a deeper look?

**Comment by hendrikebbers** [comment]
New functionality is merged in master and will be available with release 0.8.7. Since this error happens in an application that is based on DP we can not test it directly. Therefore I will move this issue to milestone 0.8.7 and we can check if the issue is fixed once the application is based on 0.8.6.

**Comment by hendrikebbers** [comment]
Maybe related to #210
